pkg/sdk/providers/amazon/arn/arn.go

```diff
@@ -0,0 +1,296 @@
+	// ARNSectionSeparator)).
+	return strings.SplitN(arn, ARNSectionSeparator, len(strings.Split(ARNSections, ARNSectionSeparator)))[4]
+}
+
+// IsARN determines whether a string represents a valid ARN object by checking
+// the ARN schema of the string.
+func IsARN(candidate string) (isARN bool) {
+	return ValidateARN(candidate) == nil
+}
+
+// NewARN creates an ARN string from the specified sections and sub-sections.
+func NewARN(
+	partition, service, region, accountID, resourceType, resourcePathOrParent, resourceName, resourceQualifier string,
+) (arn string) {
+	builder := strings.Builder{}
+	builder.WriteString(ARNSectionPrefix)
+	builder.WriteString(ARNSectionSeparator)
+	builder.WriteString(partition)
+	builder.WriteString(ARNSectionSeparator)
+	builder.WriteString(service)
+	builder.WriteString(ARNSectionSeparator)
+	builder.WriteString(region)
+	builder.WriteString(ARNSectionSeparator)
+	builder.WriteString(accountID)
+	builder.WriteString(ARNSectionSeparator)
+
+	if resourceType != "" {
+		builder.WriteString(resourceType)
+	}
+
+	builder.WriteString(resourcePathOrParent)
+	builder.WriteString(resourceName)
+
+	if resourceQualifier != "" {
+		builder.WriteString(ARNQualifierSeparator)
+		builder.WriteString(resourceQualifier)
+	}
+
+	return builder.String()
+}
+
+// Partition returns the content of the corresponding section.
+//
+// The partition that the resource is in. For standard AWS regions, the
+// partition is "aws". If you have resources in other partitions, the partition
+// is "aws-partitionname". For example, the partition for resources in the China
+// (Beijing) region is "aws-cn".
+func Partition(arn string) (partition string) {
+	if !IsARN(arn) {
+		return ""
+	}
+
+	// Note: validation ensures len(split) >= len(strings.Split(ARNSections,
+	// ARNSectionSeparator)).
+	return strings.SplitN(arn, ARNSectionSeparator, len(strings.Split(ARNSections, ARNSectionSeparator)))[1]
+}
+
+// Region returns the content of the corresponding section.
+//
+// The region the resource resides in. Note that the ARNs for some resources do
+// not require a region, so this component might be omitted.
+func Region(arn string) (region string) {
+	if !IsARN(arn) {
+		return ""
+	}
+
+	// Note: validation ensures len(split) >= len(strings.Split(ARNSections,
+	// ARNSectionSeparator)).
+	return strings.SplitN(arn, ARNSectionSeparator, len(strings.Split(ARNSections, ARNSectionSeparator)))[3]
+}
+
+// Resource returns the content of the corresponding section.
+//
+// An example resource looks like this:
+//
+// type(:|/)(parent-resource/|path/elements/)id:qualifier
+//
+// The content of this part of the ARN varies by service. It often includes an
+// indicator of the type of resource — for example, an IAM user or Amazon RDS
+// database - followed by a slash (/) or a colon (:), followed by the resource
+// name itself. Some services allows paths for resource names, as described in
```

Comment:
```suggestion
// name itself. Some services allow paths for resource names, as described in
```

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/c918bc156b0d1e596ae5ca138eba0de2027cfd59..6cc8910e94c7d0ded3ec88048287a93f7eecd69f).

Reply:
I just realized afterwards this was copied from the `github.com/aws/aws-sdk-go/aws/arn` counterpart, but it's fine anyway.